### PR TITLE
Authorization page and functionality

### PR DESCRIPTION
**Pull request by sushil-mod**
# Authorization page and functionality
how to check :- click on "login button" enter dummy provided credential and for signup click on "create new account" on login page . And from login/signup page click on "Jedi store logo" to go home page

- email :- adarshbalika@gmail.com
- password :- adarshbalika

## what to check

- AuthorizerContext :- src/context/AuthorizerContext.jsx
- Login :- src/pages/auth/Login.jsx
- Signup :- src/pages/auth/Signup.jsx

### screenshots
storelogin
storesignup

### CODE REVIEW

File: src/context/AuthorizerContext.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const AuthorizerContext = createContext();
+
+const AuthorizeProvider = ({children})=>{
+    const initialAuthState = {
+        loginStatus : false ,
+        encodedToken :""
+    }
+
+    const [ authState , authDispatch ] = useReducer( authReducerFunc , initialAuthState);
+    console.log( "from context" ,authState);
```
Comment (tusharkandpal) [inline src/context/AuthorizerContext.jsx:15]: Please remove this logger.

File: src/pages/auth/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+const loginSubmitHandler=(e)=>{
+
+    e.preventDefault();
+    const {email , password } = userInput;
+    (async () => {
+
+        try {
+            const {data , status } = await axios.post("/api/auth/login",{email,password});
+
+            console.log("from login",data , status);
```
Comment (tusharkandpal) [inline src/pages/auth/Login.jsx:24]: This too, please!

File: src/pages/auth/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+        e.preventDefault();
+
+        const {email , password ,firstName,lastName} = signupInput; 
+        (async () => {
+
+            try {
+                const {data , status} = await axios.post("/api/auth/signup",{email,password,firstName,lastName})
+
+                console.log("from signup ",  data , status);
```
Comment (tusharkandpal) [inline src/pages/auth/Signup.jsx:29]: +1

File: src/reducer/authReducerFunc.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+
+const authReducerFunc =( authState , authAction )=>{
+
+    const {type , payload }=authAction;
+    console.log("from reducer" , type , payload )
```
Comment (tusharkandpal) [inline src/reducer/authReducerFunc.jsx:7]: +1

File: src/context/AuthorizerContext.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import React ,{useReducer ,createContext ,useContext }from 'react';
+import  authReducerFunc from '../reducer/authReducerFunc';
+
+
+
+const AuthorizerContext = createContext();
+
+const AuthorizeProvider = ({children})=>{
+    const initialAuthState = {
```
Comment (tusharkandpal) [inline src/context/AuthorizerContext.jsx:9]: Can define this `initialState` in respective reducer & import it here.
Also, can use the `initialState` in reducer for case: `LOGOUT`.

**Review by ozairkhan2810 — APPROVED** [review]
Code is well divided into components and clean
Just remove console's

**Review by tusharkandpal — APPROVED** [review]
Neat ! Just suggested few changes.

File: src/context/AuthorizerContext.jsx (hunk 1 above)
Comment (sushil-mod) [reply src/context/AuthorizerContext.jsx:15]: yes forgot that

File: src/pages/auth/Login.jsx (hunk 2 above)
Comment (sushil-mod) [reply src/pages/auth/Login.jsx:24]: yes

File: src/pages/auth/Signup.jsx (hunk 3 above)
Comment (sushil-mod) [reply src/pages/auth/Signup.jsx:29]: thank you for pointing each console

File: src/reducer/authReducerFunc.jsx (hunk 4 above)
Comment (sushil-mod) [reply src/reducer/authReducerFunc.jsx:7]: yes

File: src/context/AuthorizerContext.jsx (hunk 5 above)
Comment (sushil-mod) [reply src/context/AuthorizerContext.jsx:9]: ohk i will do that ,thank you

**Pull request merged**